File: service/templates/java/WebsocketClient.java
${packageName ? 'package ' + packageName + ';' : ''}

import io.micronaut.websocket.WebSocketSession;
import io.micronaut.websocket.annotation.*;

@ClientWebSocket("/${propertyName}/{topic}/")
public abstract class ${className} implements AutoCloseable {

    private WebSocketSession session;

    @OnOpen
    public void onOpen(WebSocketSession session) {
        this.session = session;
    }

    @OnMessage
    public void onMessage() {}

    public WebSocketSession getSession() {
        return session;
    }
}

File: configuration/templates/kotlin/WebsocketClient.kt
${packageName ? 'package ' + packageName : '' }

import io.micronaut.websocket.WebSocketSession
import io.micronaut.websocket.annotation.*

@ClientWebSocket("/${propertyName}/{topic}/")
abstract class ${className} : AutoCloseable {

    var session: WebSocketSession? = null
        private set

    @OnOpen
    fun onOpen(session: WebSocketSession) {
        this.session = session
    }

    @OnMessage
    fun onMessage() {
    }
}